### ISSUE TITLE
Go snippet evolution

### ISSUE THREAD

**Issue by ccoVeille** [hougesen/mdsf#448]
You already fixed issues with Go code with my previous reported issue

- #151

I found a new pattern, but I have a new solution.

The example I used where struct, so a type definition,something that works simply by adding a package.

Unfortunately, there are piece of code, that are not formatted, and lead to errors today

Here are example

```go
f, err := strconv.Atoi("42")
if err != nil {

    panic(err) }

fmt.Println(f +3
)
```

Here the code is badly formatted on purpose.

This snippet cannot work with the following solution.

Here is what could be done

- add the same header, as you already do now
- test if it works, great
- if fails, enclose the code with this

```go
package main

func main() {

// a random string
<insert the code snippet here>
// the same random string
}
```

- try to format
- if it works, use code to extract everything between the random strings delimiter
- trim the leading space on each line returned.

**Comment by hougesen** [comment]
So if I understand it correctly the issue is that code must be inside a function definition?

The only "issue" I can see from this is that mdsf would also have to handle removing the extra indentation, which varies depending on formatter and configuration.

**Comment by ccoVeille** [comment]
Yes, exactly. I knew my suggestion is a bit messy, but I'm afraid I don't see another solution right now.

And the issue will also be that the formatting will import the path.

So this

```go
f, err := strconv.Atoi("42")
if err != nil {

    panic(err) }

fmt.Println(f +3
)
```

will have to be written like this before being sent to formatter

```go
package main

func main() {
	f, err := strconv.Atoi("42")
if err != nil {

    panic(err) }

fmt.Println(f +3,
)
}
```

https://go.dev/play/p/kMQRuIxEo-Z

would be formatted like this (please note the imports were added)

https://go.dev/play/p/xIBzjl-lUCh
```
package main

import (
	"fmt"
	"strconv"
)

func main() {
	f, err := strconv.Atoi("42")
	if err != nil {

		panic(err)
	}

	fmt.Println(f + 3)
}
```

So here you would have to extract the content of main

```go
	f, err := strconv.Atoi("42")
	if err != nil {

		panic(err)
	}

	fmt.Println(f + 3)
```

and trim the leading spaces

```go
f, err := strconv.Atoi("42")
if err != nil {
		panic(err)
}

fmt.Println(f + 3)
```

**Comment by ccoVeille** [comment]
That's why I suggested adding comment block to split the text

https://go.dev/play/p/PMDiIRjpvcV

format then split the content between

https://go.dev/play/p/4t5Ltud5EL-
